src/control/lib/netdetect/netdetect_test.go

```diff
@@ -122,3 +122,79 @@ func TestValidateNetworkConfig(t *testing.T) {
 		AssertEqual(t, err, nil, "Network device configuration is invalid - NUMA node does not match")
 	}
 }
+
+// TestDeviceAlias uses XML topology data to simulate real systems.
+// hwloc will use this topology for queries instead of the local system running the test.
+// This test verifies that GetDeviceAlias() is capable of performing this lookup
+// This particular test verifies the expected lookup path that uses a device found
+// on the system device list, and expected to find the related sibling.
+func TestDeviceAlias(t *testing.T) {
+	tests := []struct {
+		device   string
+		topology string
+		expected string
+	}{
+		{"lo", "testdata/boro-84.xml", "lo"},
+		{"eth0", "testdata/boro-84.xml", "i40iw1"},
+		{"eth1", "testdata/boro-84.xml", "i40iw0"},
+		{"ib0", "testdata/boro-84.xml", "hfi1_0"},
+		{"lo", "testdata/wolf-133.xml", "lo"},
+		{"eth0", "testdata/wolf-133.xml", "i40iw1"},
+		{"eth1", "testdata/wolf-133.xml", "i40iw0"},
+		{"ib0", "testdata/wolf-133.xml", "hfi1_0"},
+		{"ib1", "testdata/wolf-133.xml", "hfi1_1"},
+		// The following data set verifies that the sibling lookup functions properly
+		// even when the search device and expected target are swapped
+		{"i40iw1", "testdata/boro-84.xml", "eth0"},
+		{"i40iw0", "testdata/boro-84.xml", "eth1"},
+		{"hfi1_0", "testdata/boro-84.xml", "ib0"},
+		{"i40iw1", "testdata/wolf-133.xml", "eth0"},
+		{"i40iw0", "testdata/wolf-133.xml", "eth1"},
+		{"hfi1_0", "testdata/wolf-133.xml", "ib0"},
+		{"hfi1_1", "testdata/wolf-133.xml", "ib1"},
+	}
+
+	for _, tt := range tests {
+		_, err := os.Stat(tt.topology)
+		AssertEqual(t, err, nil, "unable to load xmlTopology")
+		os.Setenv("HWLOC_XMLFILE", tt.topology)
+		deviceAlias, err := GetDeviceAlias(tt.device)
+		if err != nil {
+			t.Fatal(err)
+		}
+		os.Unsetenv("HWLOC_XMLFILE")
```

Comment:
This should probably be run via defer after the Setenv() so that it it's always run at exit regardless of what happens next.

Reply:
Yes, good idea.